euphoria.cabal

```diff
@@ -97,6 +97,7 @@ library
                      , containers >= 0.5.5
                      , unordered-containers >= 0.2.5
                      , transformers >= 0.4.1
+                     , semigroups
```

Comment:
Should we apply https://ghc.haskell.org/trac/ghc/wiki/Migration/8.4#base-4.11.0.0?